opencog/query/InitiateSearchCB.cc

```diff
@@ -149,94 +149,208 @@ void InitiateSearchCB::set_pattern(const Variables& vars,
 //
 
 Handle
-InitiateSearchCB::find_starter(const Handle& h, size_t& depth,
-                                     Handle& startrm, size_t& width)
+InitiateSearchCB::find_starter(const Handle& root, size_t& depth,
+                                     Handle& link, size_t& width)
 {
 	// If its a node, then we are done.
-	Type t = h->get_type();
-	if (_nameserver.isNode(t))
+	Type root_node_type = root->get_type();
+	if (_nameserver.isNode(root_node_type))
 	{
-		if (VARIABLE_NODE != t and GLOB_NODE != t)
+		if (VARIABLE_NODE != root_node_type and GLOB_NODE != root_node_type)
 		{
-			width = h->getIncomingSetSize();
-			startrm = h; // XXX wtf ???
-			return h;
+			depth = 0;
+			width = root->getIncomingSetSize();
+			link = root; // XXX wtf ???
+			return root;
 		}
 		return Handle::UNDEFINED;
 	}
 
 	// If its a link, then find recursively
-	return find_starter_recursive(h, depth, startrm, width);
+	return find_starter_recursive(root, depth, link, width);
 }
 
-Handle
-InitiateSearchCB::find_starter_recursive(const Handle& h, size_t& depth,
-                                         Handle& startrm, size_t& width)
+Handle InitiateSearchCB::find_starter_recursive(const Handle& root, size_t& depth,
```

Comment:
Similar comment here: its clearly not the "root" any more, its just some arbitrary atom somewhere far under the root, at this point. Calling it the "root" when its not the root is .. confusing.